src/libsystemd/sd-device/sd-device.c

```diff
@@ -2655,9 +2655,7 @@ _public_ int sd_device_open(sd_device *device, int flags) {
         if (fd2 < 0)
                 return fd2;
 
-        if (diskseq == 0)
-                return TAKE_FD(fd2);
-
+        /* If diskseq isn't supported it should remain unsupported.  */
```

Comment:
this is very cryptic grok outside of the context of this very patch. Please say explicitly that in case the device doesn't support diskseq we'll see a zero diskseq here, and that shouldn't change over the lifetime of the device.